crates/vault-registry/src/lib.rs

```diff
@@ -641,10 +643,29 @@ impl<T: Config> Pallet<T> {
             Self::is_allowed_to_withdraw_collateral(vault_id, amount)?,
             Error::<T>::InsufficientCollateral
         );
-
+        ensure!(
+            Self::is_max_nomination_ratio_preserved(vault_id, amount)?,
+            Error::<T>::MaxNominatioRatioViolation
+        );
         Self::force_withdraw_collateral(vault_id, amount)
     }
 
+    pub fn is_max_nomination_ratio_preserved(
+        vault_id: &T::AccountId,
+        amount: Collateral<T>,
+    ) -> Result<bool, DispatchError> {
+        let vault_collateral = Self::compute_collateral(vault_id)?;
+        let backing_collateral = Self::get_backing_collateral(vault_id)?;
+        let current_nomination = backing_collateral
+            .checked_sub(&vault_collateral)
+            .ok_or(Error::<T>::ArithmeticUnderflow)?;
+        let new_vault_collateral = vault_collateral
+            .checked_sub(&amount)
+            .ok_or(Error::<T>::ArithmeticUnderflow)?;
+        let max_nomination_after_withdrawal = Self::get_max_nominatable_collateral(new_vault_collateral)?;
+        Ok(current_nomination.le(&max_nomination_after_withdrawal))
```

Comment:
stylistic only, but in most of our code we use 
```suggestion
        Ok(current_nomination <= max_nomination_after_withdrawal)
```